step/step.go

```diff
@@ -1,42 +1,91 @@
 package step
 
 import (
+	"encoding/json"
 	"fmt"
 	"os"
-)
 
-const (
-	ExitCodeOK      = 0
-	ExitCodeFailure = 1
+	"github.com/Jeffail/gabs/v2"
+	"github.com/stackpulse/steps-sdk-go/env"
 )
 
 type Outputs struct {
 	Object interface{} `json:"api_object,omitempty"`
 }
 
+type JSONOutput interface{}
+
 // An interface describing a general purpose stackpulse step
 type Step interface {
-	Init() error
-	Run() (exitCode int, output []byte, err error)
+	Init() *Error
+	Run() (outputJson JSONOutput, stepError *Error)
```

Comment:
named parameters are not needed, its self explanatory
```suggestion
	Run() JSONOutput, stepError *Error
```